### PR TITLE
Improve documentations (see #354)

### PR DESCRIPTION
**Pull request by jakwings**
ping #354

### CODE REVIEW

File: doc_src/index.hdr.in [hunk 1]
```diff
@@ -498,6 +496,32 @@ end
...
+echo $a$b
+# Outputs 'x1 y1 z1 x2 y2 z2 x3 y3 z3'
+echo $a"-"$b
+# Outputs 'x-1 y-1 z-1 x-2 y-2 z-2 x-3 y-3 z-3'
+
+echo {x,y,z}$b
+# Outputs 'x1 y1 z1 x2 y2 z2 x3 y3 z3'
+
+echo {$a}word
+# Outputs '1word 2word 3word'
```
Comment (faho) [inline doc_src/index.hdr.in]: xword yword zword, no?

**Comment by faho** [comment]
Regarding the first commit: These two things behave differently. Quotes will pass variables as one argument (i.e. won't trigger cartesian products), while braces will. I'd rather mention both styles than completely replace it with the quote-version.

Regarding the second: We already have a bit on cartesian products in the tutorial page - presumably these two should at the least be linked.

File: doc_src/index.hdr.in (hunk 1 above)
Comment (jakwings) [inline doc_src/index.hdr.in]: Thanks. fixed

**Comment by jakwings** [comment]
> Regarding the first commit: These two things behave differently. Quotes will pass variables as one argument (i.e. won't trigger cartesian products), while braces will. I'd rather mention both styles than completely replace it with the quote-version.

If users has know them both well, they will know how to use them. `{"$var"}` is a complex use.

> Regarding the second: We already have a bit on cartesian products in the tutorial page - presumably these two should at the least be linked.

Linked.

**Comment by faho** [comment]
>If users has known them both well, they will know how to use them. {"$var"} is a complex use.

Sorry, I should have been more clear - I mean `echo {$var}banana` and `echo "$var"banana` should _both_ be mentioned.

**Comment by jakwings** [comment]
Fine, the original example added back.

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**